### PR TITLE
[TIMOB-7970] Remove offending ICU calls

### PR DESCRIPTION
**Pull request by sptramer**
Apple validation was indicating that TiCore was calling out to some "private" methods inside the ICU library. This fix interfaces with them in the approved manner through CFLocale and CFString.

As a bonus there were some fixes made to the fixup script and building the XCode project without buildit.

### CODE REVIEW

File: TiCore/wtf/unicode/icu/CollatorICU.cpp [hunk 1]
```diff
@@ -98,14 +109,48 @@ void Collator::setOrderLowerFirst(bool lowerFirst)
...
+    RetainPtr<CFLocaleRef> locale = CFLocaleCreate(NULL, localeStr.get());
+    
+    if (!locale) {
+        locale = CFLocaleCopyCurrent();
+    }
+    
+    // CFStringCreateWithCharacters does not accept -1; we have to determine the terminator position ourselves.
+    if (lhsLength == (size_t)(-1)) {
+        lhsLength = 0;
+        while (*(lhs++) != 0x0000) {
```
Comment (WhichKatieDid) [inline TiCore/wtf/unicode/icu/CollatorICU.cpp]: Talked with stramer on this one over IRC.

**Comment by WhichKatieDid** [comment]
Tentative reject, to reevaluate pending fix for the pointer movement.

**Comment by dawsontoth** [comment]
FR **passed**. Was able to validate through Xcode's "Validate..." test successfully.

**Comment by WhichKatieDid** [comment]
Code review APPROVED. Doublechecked TiJSCore, and code is only invoked once, and lengths are never -1.

**Pull request merged**